liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1464,14 +1464,31 @@ private void doDevMode() throws MojoExecutionException {
         final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
                 new ArrayBlockingQueue<Runnable>(1, true));
 
+        sourceDirectory = new File(sourceDirectoryString.trim());
+        testSourceDirectory = new File(testSourceDirectoryString.trim());
+
+        ArrayList<File> javaFiles = new ArrayList<File>();
+        listFiles(sourceDirectory, javaFiles, "java");
+
+        ArrayList<File> javaTestFiles = new ArrayList<File>();
+        listFiles(testSourceDirectory, javaTestFiles, "java");
+
+        getLog().debug("Source directory: " + sourceDirectory);
+        getLog().debug("Output directory: " + outputDirectory);
+        getLog().debug("Test Source directory: " + testSourceDirectory);
+        getLog().debug("Test Output directory: " + testOutputDirectory);
+
         if (isEar) {
             runMojo("org.apache.maven.plugins", "maven-ear-plugin", "generate-application-xml");
             runMojo("org.apache.maven.plugins", "maven-resources-plugin", "resources");
-            // for test classes in ear
-            try {
-                runCompileMojoLogWarningWithException("testCompile");
-            } catch (MojoExecutionException e) {
-                compileMojoError.put(project.getName(),Boolean.TRUE);
+            if(!javaTestFiles.isEmpty()) {
+                runMojo("org.apache.maven.plugins", "maven-resources-plugin", "testResources");
```

Comment:
was this part missing before? I see it below for `war`. Just want to make sure it applies to `ear` also.